### PR TITLE
Updating BUSCO to 5.8.0

### LINKED ISSUE #6496: Updating BUSCO to 5.8.0 for bug fixes
Multiple users have reported a recurring bug that aligns with an issue identified upstream as reported [here](https://gitlab.com/ezlab/busco/-/issues/767).

To address this, we recommend updating BUSCO from version 5.7.1 to 5.8.0. The new tool update contains necessary bug fixes that should resolve this recurring issue for our users.

### PR DESCRIPTION
**Pull request by SaimMomin12**
FOR CONTRIBUTOR:
* [x] I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] License permits unrestricted use (educational + commercial)
* [ ] This PR adds a new tool or tool collection
* [x] This PR updates an existing tool or tool collection
* [ ] This PR does something else (explain below)

---
This PR updates BUSCO and associated test data to its latest version 5.8.0 and fixes #6496

### CODE REVIEW

File: tools/busco/busco.xml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <tool id="busco" name="Busco" profile="20.01" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@">
```
Comment (bgruening) [inline tools/busco/busco.xml:1]:
```suggestion
<tool id="busco" name="Busco" profile="20.01" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="23.0">
```

File: tools/busco/test-data/genome_results_miniprot/out.gff
Comment (bgruening) [inline tools/busco/test-data/genome_results_miniprot/out.gff:1]: @SaimMomin12 why is this file been added here?

Comment (SaimMomin12) [reply tools/busco/test-data/genome_results_miniprot/out.gff:1]: I guess, this file already existed before updating the tool. I did not add explicitly.

Comment (SaimMomin12) [reply tools/busco/test-data/genome_results_miniprot/out.gff:1]: It comes from a test as described here: https://github.com/galaxyproject/tools-iuc/pull/6497/files#diff-9e60d1215fc56adcc3de9e54a57a93d0ec5300216b7c5d438d6ab34c8a41b1eaL403

**Pull request merged**